src/cephadm/cephadm

```diff
@@ -5212,6 +5260,507 @@ def command_gather_facts():
+                        data = json.loads(stdout)
+                    except json.decoder.JSONDecodeError:
+                        errors.append("ceph-volume thread provided bad json data")
+                        logger.warning(errors[-1])
+                        data = []
+                else:
+                    errors.append("ceph-volume didn't return any data")
+                    logger.warning(errors[-1])
+
+                self.cephadm_cache.update_task(
+                    'disks',
+                    {
+                        "scrape_timestamp": s_time,
+                        "scrape_duration_secs": elapsed,
+                        "scrape_errors": errors,
+                        "data": data, 
+                    }
+                )
+                
+                logger.debug(f"completed ceph-volume scrape - {elapsed}s")
+            time.sleep(CephadmDaemon.loop_delay)
+            ctr += CephadmDaemon.loop_delay
+        
+        logger.info("ceph-volume thread stopped")
+
+    def _scrape_list_daemons(self, refresh_interval=20):
+        ctr = 0
+        while True:
+            if self.stop:
+                break
+            
+            if ctr >= refresh_interval:
+                ctr = 0
+                logger.debug("executing list-daemons scrape")
+                errors = []
+                s_time = time.time()
+
+                # list daemons should ideally be invoked with a fsid
+                data = list_daemons()
+                if not isinstance(data, list):
+                    errors.append("list-daemons didn't supply a list?")
+                    logger.warning(errors[-1])
+                    data = []
+                elapsed = time.time() - s_time
+                self.cephadm_cache.update_task(
+                    'daemons',
+                    {
+                        "scrape_timestamp": s_time,
+                        "scrape_duration_secs": elapsed,
+                        "scrape_errors": errors,
+                        "data": data,
+                    }
+                )
+                logger.debug(f"completed list-daemons scrape - {elapsed}s")
+            
+            time.sleep(CephadmDaemon.loop_delay)
+            ctr += CephadmDaemon.loop_delay
+        logger.info("list-daemons thread stopped")
+
+    def _create_thread(self, target, name, refresh_interval=None):
+        if refresh_interval:
+            t = Thread(target=target, args=(refresh_interval,))
+        else:
+            t = Thread(target=target)
+        t.daemon = True
+        t.name = name
+        self.cephadm_cache.update_health(name, "active")
+        t.start()
+
+        start_msg = f"Started {name} thread"
+        if refresh_interval:
+            logger.info(f"{start_msg}, with a refresh interval of {refresh_interval}s")
+        else:
+            logger.info(f"{start_msg}")
+        return t
+
+    def reload(self, *args):
+        logger.info("Reload request received")
```

Comment:
Hm.. Looks like the reload call is missing in the reload method.

Reply:
Yeah - I'd put it there as a placeholder, but not all the plumbing was in place. Just added another patch that completes the reload plumbing, so the exporter can react to a systemctl reload command.